swap/log.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package swap
+
+import (
+	log "github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethersphere/swarm/network"
+)
+
+const (
+	// InitAction used when starting swap
+	InitAction string = "init"
+	// StopAction used when stopping swap
+	StopAction string = "stop"
+	// UpdateBalanceAction used when updating swap balances
+	UpdateBalanceAction string = "update_balance"
+	// SendChequeAction used for grouping actions when sending a cheque with swap
+	SendChequeAction string = "send_cheque"
+	// HandleChequeAction used for grouping actions related to swap cheque events, received/processed/etc cheques
+	HandleChequeAction string = "handle_cheque"
+	// CashChequeAction used for grouping actions of swap cashed cheques
+	CashChequeAction string = "cash_cheque"
+	// DeployChequebookAction used when deploying chequebooks
+	DeployChequebookAction string = "deploy_chequebook_contract"
+)
+
+// DefaultSwapLogLevel indicates default filter level of log messages
+const DefaultSwapLogLevel = 3
+
+const fileSizeLimit = 262144 // max bytes limit for splitting file in parts
+const emptyLogPath = ""      // Used when no logPath is specified for a logger
+
+// Logger wraps the ethereum logger with specific information for swap logging
+// each log contains a context which will be printed on each message
+type Logger struct {
```

Comment:
I wonder if the logger here could be moved to a separate package (inside `swap` package), so that it can also be used from the `swap/chain` package without circular dependency problems.

Reply:
There is a branch `swap-logger-actions-updated-package`
That contains a refactor of the swapLog into package.

But It would require to reference the actions thru example `swapLog.InitAction`

We can maybe close this PR as is and when the chain package is refactored, add this as well.

Any opinion @mortelli ?

Reply:
I would keep the actions (like `InitAction`) inside the swap package as the actions are package specific. But the `Logger` struct and all the functions (except `newPeerLogger` and alike) should be in a separate package. But I can make that move when rebasing. From my side you can leave the PR as is, just figure out the `go.sum` issue.

Reply:
That's a good idea, we should address it later then.

I fixed the `go.sum` with your command.
Thanks for the feedback.